app/routes.py

```diff
@@ -1,2 +1,211 @@
+
+    if request.method == 'GET':
+        return jsonify(video.to_json()),200
+
+    if request.method == 'PUT':
+        try:
+            video.title = request_body['title'],
+            video.release_date = request_body['release_date'],
+            video.total_inventory = request_body['total_inventory']
+            db.session.commit()
+            return video.to_json(), 200
+        except KeyError:
+            return make_response('{}', 400)
+
+    if request.method == 'DELETE':
+        db.session.delete(video)
+        db.session.commit()
+        return make_response(video.to_json(), 200)
+
+
+
+@video_bp.route('/<id>/rentals', methods=['GET'])
+def video_rentals(id):
+    video = Video.query.filter_by(id=id).first()
+    response_list = []
+    for rental in video.rentals:
+        d = {
+            'due_date': rental.due_date,
+            'name': rental.customer.name,
+            'phone': rental.customer.phone,
+            'postal_code': rental.customer.postal_code
+        }
+        response_list.append(d)
+    return jsonify(response_list), 200
+
+
+# Rentals
+
+@rental_bp.route('/check-out', methods=['POST'])
+def rental_checkout():
+    request_body = request.get_json()
+    video_id = request_body.get('video_id')
+    customer_id = request_body.get('customer_id')
+    if not type(customer_id) == int or not type(video_id) == int:
+        return make_response({'details': 'Invalid data'}, 400)
+    video = Video.query.filter_by(id=video_id).first()
+    if not video.available_inventory:
+        return {'Details': 'No inventory available'}, 400
+    
+    try:
+        rental = Rental(customer_id=request_body['customer_id'],
+                        video_id=request_body['video_id'],
+                        due_date=str(datetime.today().date() + timedelta(days=7)))
+        db.session.add(rental)
+        db.session.commit()
+        response = {
+            'customer_id': rental.customer_id,
+            'video_id': rental.video_id,
+            'due_date': rental.due_date,
+            'videos_checked_out_count': rental.customer.videos_checked_out_count,
+            'available_inventory': rental.video.available_inventory,
+        }
+        return response, 200
+    except KeyError:
+            return "{}", 400
+
+
+@rental_bp.route('/check-in', methods=['POST'])
+def rental_checkin():
+    request_body = request.get_json()
+    video_id = request_body.get('video_id')
+    customer_id = request_body.get('customer_id')
+    if not type(customer_id) == int or not type(video_id) == int:
+        return make_response({'details': 'Invalid data'}, 400)
+    try:
+        video = Video.query.filter_by(id=video_id).first()
+        customer = Customer.query.filter_by(id=customer_id).first()
+        rental_queryset = Rental.query.filter_by(video_id=video_id, customer_id=customer_id)
+        if not rental_queryset.count():
+            return {'Details': 'Nope!'}, 400
+        rental_queryset.delete()
```

Comment:
Just note that by deleting the Rental, you lose all records of it.  So if you wanted to find out who last rented that copy of "The Princess Bride," it wouldn't exist.